tests/Keywords/DialectKeywordsTest.php

```diff
@@ -35,26 +36,23 @@ protected static function getKeywords(): KeywordsInterface
 
     protected static function getKeywordsArray(): array
     {
-        $data = Filesystem::readJsonFileArray(__DIR__ . '/../../resources/gherkin-languages.json');
-
-        $keywordsArray = [];
-
-        foreach ($data as $lang => $dialect) {
-            $keywordsArray[$lang] = [
-                'feature' => implode('|', $dialect['feature']),
-                'background' => implode('|', $dialect['background']),
-                'scenario' => implode('|', $dialect['scenario']),
-                'scenario_outline' => implode('|', $dialect['scenarioOutline']),
-                'examples' => implode('|', $dialect['examples']),
-                'given' => implode('|', $dialect['given']),
-                'when' => implode('|', $dialect['when']),
-                'then' => implode('|', $dialect['then']),
-                'and' => implode('|', $dialect['and']),
-                'but' => implode('|', $dialect['but']),
-            ];
-        }
-
-        return $keywordsArray;
+        return array_map(
+            static fn ($dialect) => is_array($dialect)
+                ? [
+                    'feature' => implode('|', (array) ($dialect['feature'] ?? [])),
+                    'background' => implode('|', (array) ($dialect['background'] ?? [])),
+                    'scenario' => implode('|', (array) ($dialect['scenario'] ?? [])),
+                    'scenario_outline' => implode('|', (array) ($dialect['scenarioOutline'] ?? [])),
+                    'examples' => implode('|', (array) ($dialect['examples'] ?? [])),
+                    'given' => implode('|', (array) ($dialect['given'] ?? [])),
+                    'when' => implode('|', (array) ($dialect['when'] ?? [])),
+                    'then' => implode('|', (array) ($dialect['then'] ?? [])),
+                    'and' => implode('|', (array) ($dialect['and'] ?? [])),
+                    'but' => implode('|', (array) ($dialect['but'] ?? [])),
+                ]
+                : throw new RuntimeException('Invalid data; expected array of keywords but got: ' . get_debug_type($dialect)),
+            Filesystem::readJsonFileHash(__DIR__ . '/../../resources/gherkin-languages.json'),
+        );
```

Comment:
Rather than changing this implementation, is it possible to solve it by telling PHPstan the JSON has the (known) structure as in the CucumberDialectProvider e.g. 

https://github.com/Behat/Gherkin/blob/1325278942943332a6bcdc02b99fab76fa20bfb4/src/Dialect/CucumberDialectProvider.php#L30-L36

I see we also do that in the TranslationTest below.

Or does that not work in this context?

Reply:
Considering that this is a test data provider, I suppose it's ok to ignore the warnings (and let if fail at runtime, in case some problems comes up from the upstream cucumber repo).

Reply:
PS: `$data` is not compatible with `TDialectData`, e.g. `scenarioOutline` vs `scenario_outline`

Reply:
That would also be fine, the JSON file can only change in the job that updates to a newer upstream and if that fails it will be fairly clear what's going on. Plus I assume their dialect schema is fairly well fixed now.

Reply:
I've applied an update - please reopen this for further feedback.